src/main/java/com/artipie/asto/fs/RxFile.java

```diff
@@ -100,4 +101,17 @@ public Completable save(final Flowable<ByteBuffer> flow) {
                 )
             ).delay(delay, TimeUnit.MILLISECONDS);
     }
+
+    /**
+     * Move file to new location.
+     * @param target Target path the file is moved to.
+     * @return Completion or error signal
+     */
+    public Completable move(final Path target) {
```

Comment:
I would prefer to avoid implementing `move` in `RxFile`, instead, I would use a combination of `save` and `flow` method in `FileStorage`, doing something similar to:
`new RxFile(dest).save(new RxFile(source).flow())`

Reply:
Why is that? To me one of important points in having `move()` method is the possibility to this operation efficiently. In most storages (file system, s3) moving data will be fast operation which complexity does not depend on file size. Just copying all the bytes will cost much more resources.

Reply:
You are right.